typescript/cli/src/utils/balances.ts

```diff
@@ -40,18 +40,16 @@ export async function gasBalancesAreSufficient(
   minGas: string,
 ): Promise<boolean> {
   let sufficient = true;
-  await Promise.all(
-    chains.map(async (chain) => {
-      const provider = multiProvider.getProvider(chain);
-      const gasPrice = await provider.getGasPrice();
-      const minBalanceWei = gasPrice.mul(minGas).toString();
-      sufficient = await nativeBalancesAreSufficient(
-        multiProvider,
-        signer,
-        [chain],
-        minBalanceWei,
-      );
-    }),
-  );
+  for (const chain of chains) {
+    const provider = multiProvider.getProvider(chain);
+    const gasPrice = await provider.getGasPrice();
+    const minBalanceWei = gasPrice.mul(minGas).toString();
+    sufficient = await nativeBalancesAreSufficient(
+      multiProvider,
+      signer,
+      [chain],
+      minBalanceWei,
+    );
```

Comment:
also now realizing this only effectively checks the last chain...

Reply:
if any of the many chains have insufficient balance, the `sufficient` flag becomes `false`– why does this only check last chain?

Reply:
if `chains.length == 2 && isSufficient(chains[0]) == false && isSufficient(chains[1]) == true` then sufficient is true after this loop ends?

Reply:
this is also true for the `nativeBalancesAreSufficient` helper, which also should not prompt and assign to a single variable across concurrent promises